test/Sql/Platform/Mysql/SelectDecoratorTest.php

```diff
@@ -95,10 +96,74 @@ public function dataProvider()
         $expectedParams2 = ['offset' => '10000000000000000000', 'limit' => '5'];
         $expectedSql2 = 'SELECT `foo`.* FROM `foo` LIMIT 5 OFFSET 10000000000000000000';
 
+        // nested single limit & offset in field param
+        $nestedSelect0 = new Select;
+        $nestedSelect0->from('foo1')
+            ->columns([
+                'cnt' => new Expression('count(foo1.id)')
+            ])->limit(0)->offset(5);
+
+        $select3 = new Select;
+        $select3->from('foo')
+            ->columns([
+                'res'  => $nestedSelect0,
+            ])
+            ->limit('10')->offset('5');
```

Comment:
I would set here different offset, now we have the same as in the nested query, and it's possible that something is getting mixed, so to make it clear I would change the value.